src/Contract/Types/Plutus/AuctionInfo.purs

```diff
@@ -105,17 +178,18 @@ instance Show AuctionInfoValidationError where
   show = genericShow
 
 validateAuctionInfo
```

Comment:
On a general note: I would prefer this function to do parsing, not validation, so we can be sure we are not using unchecked AuctionInfo at some random point. In the extreme case, the source type can be an opaque one, preventing accessing its components directly, and the output type can be opened.